sound/soc/sof/topology.c

```diff
@@ -1383,6 +1386,8 @@ int sof_load_pipeline_ipc(struct snd_sof_dev *sdev,
 	if (ret < 0)
 		dev_err(sdev->dev, "error: core enable ipc failure\n");
 
+	mutex_unlock(&sdev->cores_mask_mutex);
```

Comment:
I think you can release the mutex much earlier - after line 1369:
```
	sdev->enabled_cores_mask |= 1 << pipeline->core;
```

Reply:
As stated, I want to wait ack from FW and aligned with FW about this power status with cores.